>**Printing operations in Python.**

In [1]:
print('Hello World')

Hello World


Python prints a _newline, by default_. Below example shows multiline printing operation.

In [2]:
print('Hello World')
print('Hello People')

Hello World
Hello People


To avoid printing _newline_, `end` parameter can be used like below.

In [3]:
print('Hello World', end = ' ')
print('Hello People')

Hello World Hello People


`end` parameter can be used to print something _in between 2 printing operations_.

In [4]:
print('1st print operation.', end = ' In between two printing operations. ')
print('2nd printing operation.')

1st print operation. In between two printing operations. 2nd printing operation.


>**Variables in Python.**

In [5]:
name = 'Mr. Roudolph Lewis'
age = 63
height = 5.5
isMale = True

print('Name:', name)
print('Age:', age, 'yrs')
print('Height:', height, 'ft')
print('Male:', isMale)

Name: Mr. Roudolph Lewis
Age: 63 yrs
Height: 5.5 ft
Male: True


These are the basic data-types in Python. `name` variable has been used to store _string_ value `'Mr. Roudolph Lewis'`, `age` variable has been used to store _integer_ value `63`, `height` variable has been used to store _float_ value `5.5` and lastly, `isMale` variable has been used to store _boolean_ value `True`.

_Strings_ are collection of characters, written in between quotation marks. Single `''` or double `""` quotation, either can be used. _Integers_ are just _integer_ numbers. _Floats_ are numbers having values after decimal point. And _booleans_ are just `True`/`False`.

We can name our variables just the way we want. But we have to keep in mind that we cannot start our variable names with numbers, such that: `1name`, `1age2` or `12`. These are examples of _invalid variable naming_. Rather we can use numbers or digits at the middle or at the end, like: `name1`, `heig1t2` or `a1b2c3` are _totally valid variable names_. 

We should also remember that we cannot use _keywords_ or _basic operators_ in our variable names. `a*b`, `for`, `if`, `def`, `new` or `name/` are some _invalid variable names_ as these names causes confusion.

>**Comments in Python.**

In [6]:
# This is a single-line comment in python
print('Hello World')

Hello World


In [7]:
"""
This is a multiline comment in python.
It can span multiple lines.
"""
print('Hello World')

Hello World


>**Type-conversions in Python.**

In [8]:
name = 'Donald Trump'
print(type(name))   # Prints the data-type, the variable is containing

<class 'str'>


We often convert data-types of variable in Python. _String_ to _integer_, _integer_ to _string_ or _float_ to _string_ etc.

In [9]:
value_in_string = '13'
print(type(value_in_string))
value_in_int = int(value_in_string)   # converted from string to int
print(type(value_in_int))

<class 'str'>
<class 'int'>


In [10]:
value_in_string = '55.56'
print(type(value_in_string))
value_in_float = float(value_in_string) # converted from string to float
print(type(value_in_float))

<class 'str'>
<class 'float'>


In [11]:
value_in_string = 'False'
print(type(value_in_string))
value_in_bool = bool(value_in_string)   # converted from string to boolean
print(type(value_in_bool))

<class 'str'>
<class 'bool'>


In [12]:
value_in_int = 15
print(type(value_in_int))
value_in_string = str(value_in_int) # converted from int to string
print(type(value_in_string))

value_in_float = 3.141568
print(type(value_in_float))
value_in_string = str(value_in_float) # converted from float to string
print(type(value_in_string))

value_in_bool = True
print(type(value_in_bool))
value_in_string = str(value_in_bool) # converted from bool to string
print(type(value_in_string))

<class 'int'>
<class 'str'>
<class 'float'>
<class 'str'>
<class 'bool'>
<class 'str'>


>**Input operations in Python.**

In [20]:
name = input()  # Joe
print('Hello', name)

Hello Joe


Here, the code asks for a data from the user and greets him with a _hello_. We can take inputs from user, and _store_ it in a _variable_. And then, variables can be used as our convenience. Now, if we want to print a message for the user, before input, awaring him what he should give as input, we can do as following:

In [21]:
name = input('Please enter your name:')
print('Hello', name)

Hello Joe


Now when we are going to run the code, `Please enter your name: ` message will be printed and than input for name will be taken.

In [22]:
name = input('Please enter your name:')
age = input('Please enter your age:')
height = input('Please enter your height:')

print('Your name:', name)
print('Your age:', age, 'yrs')
print('Your height:', height, 'ft')

Your name: Roudolph Lewis
Your age: 63 yrs
Your height: 5.9 ft


>**Operators in Python.**

In [16]:
a = 84
b = 12

print('Summation:', a + b)
print('Subtraction:', a - b)
print('Multiplication:', a * b)
print('Division:', a / b)
print('Modulo:', a % b)

Summation: 96
Subtraction: 72
Multiplication: 1008
Division: 7.0
Modulo: 0


`+` operator is used to perform _summation_ between two numbers, `-` operator is used to perform _subtraction_ between two numbers, `*` operator is used to perform _multiplication_ between two numbers, `/` operator is used to perform _division_ between two numbers and lastly, `%` operator is used to print _modular operation_.

Now let's try these _operators with inputs_.

In [24]:
a = input()
b = input()

sum = a+b

print('Summation of', a, 'and', b, 'is:', sum)

Summation of 10 and 12 is: 1012


Something is wrong! Summation of `10` and `12` _cannot be_ `1012`. Let's try printing the `type`s of `a` and `b` variable.

In [25]:
a = input()
b = input()

print(type(a))
print(type(b))

<class 'str'>
<class 'str'>


So here is the issue, when we take some sort of inputs, Python by default takes input as _strings_. But we can easily _convert_ the `type`s and _perform operations_ as our convenience. 

In [26]:
a = int (input())   # input string converted into int
b = int (input())   # input string converted into int

sum = a+b

print('Summation of', a, 'and', b, 'is:', sum)

Summation of 10 and 12 is: 22


Now it's working as it should.

In [ ]:
# Exponentiation Operator
a = 2
e = 10

value = a ** e  # a to the power e

print(value)

1024


Python follows conventional operator precedence so _it's a good practice to use enough brackets to solve complex equations_.

In [ ]:
a = 2
b = 3

c = a + b ** 2  # finds b to the power 2 first then performs summation with a
d = (a + b) ** 2    # finds a + b first then performs (a + b) to the power 2

print(c)
print(d)

11
25


>**Code to calculate BMI.**

In [1]:
weight = float (input('Please enter your weight in kgs:'))
height = float (input('Please enter your height in ft:'))

bmi = weight / (height ** 2)

# Prints bmi in 3 digits after decimal value
print('Your BMI is', round(bmi, 3), 'kg per ft square') 

Your BMI is 2.229 kg per ft square
